### ISSUE TITLE
Range error [offset > 6], while parsing POPM frame in ID3v2 header 

### ISSUE THREAD

**Issue by b3u** [Borewit/music-metadata#229]
**Bug description**
I'm trying to read data from an array of file paths.
I get the following error:
```sh
(node:15553) UnhandledPromiseRejectionWarning: RangeError [ERR_OUT_OF_RANGE]: The value of "offset" is out of range. It must be >= 0 and <= 6. Received 9
    at boundsError (internal/buffer.js:49:9)
    at Buffer.readUInt8 (internal/buffer.js:137:5)
    at Function.readData (/home/binyamin/Documents/Coding/musia/node_modules/music-metadata/lib/id3v2/FrameParser.js:129:31)
    at Function.readFrameData (/home/binyamin/Documents/Coding/musia/node_modules/music-metadata/lib/id3v2/ID3v2Parser.js:90:46)
    at ID3v2Parser.parseMetadata (/home/binyamin/Documents/Coding/musia/node_modules/music-metadata/lib/id3v2/ID3v2Parser.js:183:40)
    at ID3v2Parser.parseId3Data (/home/binyamin/Documents/Coding/musia/node_modules/music-metadata/lib/id3v2/ID3v2Parser.js:139:32)
(node:15553) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:15553) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Expected behavior**
Receive metadata
**Problematic Audio**

[audio.zip](https://github.com/Borewit/music-metadata/files/3465352/audio.zip)

**Comment by Borewit** [comment]
Seems to be an issue parsing the POPM field in the ID3 header.

**Comment by Borewit** [comment]
Related PR: #230

Can you try [4.2.2](https://www.npmjs.com/package/music-metadata/v/4.2.2) released as **beta**?

```shell
npm install music-metadata@beta
```

**Comment by b3u** [comment]
Great! Now I got a problem from a different file:
```sh
RangeError [ERR_INVALID_OPT_VALUE]: The value "-8" is invalid for option "size"
    at Function.alloc (buffer.js:259)
    at FileTokenizer.readToken (AbstractTokenizer.js:10)
    at MP4Parser.js:130
    at Atom.readData (Atom.js:71)
    at Atom.readAtom (Atom.js:46)
    at async Atom.readAtoms (Atom.js:18)
```

I would attach, the file, but this file is purchased so I don't feel comfortable uploading it. I can send you metadata though, if that would help.

**Comment by Borewit** [comment]
Please open a new issue for that one, otherwise my administration becomes a mess.

I am traveling at the moment, and my laptop battery is empty. Therefor that issue may need to wait a bit longer to be solved.

**Comment by Borewit** [comment]
I think I implemented the wrong solution in PR #230. 

The e-mail address fields of the POPM field is set to  ‘0x01, 0x00’ with UTF-16 encoding. I think that should have been ‘0x00, 0x00’ for an empty null terminated UTF-16 stream.

I suspect the issue is caused by a corrupt POPM field. If that is indeed the case I will prevent it from crashing entirely.

**Comment by Borewit** [comment]
Related documentation: 
* http://id3.org/id3v2.3.0#ID3v2_frame_overview
* http://id3.org/id3v2.3.0#Popularimeter
